runtime/gc_include/ObjectAllocationAPI.hpp

```diff
@@ -274,9 +327,9 @@ class MM_ObjectAllocationAPI
 		j9object_t instance = NULL;
 #if defined(J9VM_GC_THREAD_LOCAL_HEAP) || defined(J9VM_GC_SEGREGATED_HEAP)
 		/* Calculate the size of the object */
-		UDATA const headerSize = J9VMTHREAD_OBJECT_HEADER_SIZE(currentThread);
-		UDATA dataSize = clazz->totalInstanceSize;
-		UDATA allocateSize = (dataSize + headerSize + _objectAlignmentInBytes - 1) & ~(UDATA)(_objectAlignmentInBytes - 1);
+		uintptr_t const headerSize = J9VMTHREAD_OBJECT_HEADER_SIZE(currentThread);
+		uintptr_t dataSize = clazz->totalInstanceSize;
+		uintptr_t allocateSize = (dataSize + headerSize + _objectAlignmentInBytes - 1) & ~(uintptr_t)(_objectAlignmentInBytes - 1);
```

Comment:
I think this round-up pattern deserves to be refactored into a helper function (perhaps a macro already exists).